include/cuco/detail/static_multimap/kernels.cuh

```diff
@@ -468,7 +474,7 @@ template <uint32_t block_size,
           typename viewT,
           typename PairEqual>
 __global__ void pair_retrieve(InputIt first,
-                              InputIt last,
+                              int64_t n,
```

Comment:
So internally we move away from a pure iterator interface in favor of ye'olden C-style pointer+size approach? Is this because we need to evaluate `if (it < size)` where `operator <` isn't part of the strict definition of an iterator?

Reply:
It's mainly because `iter + i` can be undefined behavior. See https://github.com/rapidsai/cudf/issues/12108#issuecomment-1309503258. So we should avoid using iterator operations to determine loop scopes in future implementations.

Reply:
Thanks for the clarification. Yeah,I recall having a similar discussion in one of my previous PRs but somehow forgot to escalate it.
Will keep this in mind for the refactor.

Reply:
On second thought I feel having a pair `Iterator, int64_t` to describe a range doesn't sound right since we can't just use every iterator type. Would a plain old pointer+size combination work, i.e. `T*` and the tparam would be `T`?

Reply:
But then again, it's probably not worth it arguing over such conventions that aren't functionality related since the implementation gets replaced soonish anyways. So please ignore me 😅

Reply:
> I feel having a pair Iterator, int64_t to describe a range doesn't sound right since we can't just use any iterator type. 

I won't worry too much about this since kernel functions are in detail namespace. Our public APIs are still taking `Iterator begin` and `Iterator end`.